ffi/blitbuffer.lua

```diff
@@ -1826,6 +1826,216 @@ function BB_mt.__index:writePNGFromBGR(filename)
     C.free(cdata)
 end
 
+function BB_mt.__index:writeBMP(filename)
+    local function write_uint32(target_ptr, target_pos, data)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = data
+        return target_pos + 1
+    end
```

Comment:
Can be rewritten in a much more C-like fashion:

```
*ptr++ = (data & 0xFF);
*ptr++ = ((data >> 8) & 0xFF);
*ptr++ = ((data >> 16) & 0xFF);
*ptr++ = ((data >> 24) & 0xFF);
```

Since pointer derefs use array notation in Lua/FFI, this should turn into something like

```
target_ptr[target_pos] = band(data, 0xFF)
target_ptr[target_pos + 1] = band(rshift(data, 8), 0xFF)
target_ptr[target_pos + 2] = band(rshift(data, 16), 0xFF)
target_ptr[target_pos + 3] = band(rshift(data, 24), 0xFF)
return target_pos + 4
```

Reply:
Much better.